### DISCUSSION TITLE
No PTT when receiving and external mike is used (only for repeaters)

### DISCUSSION THREAD

**Discussion by ldogge** [egzumer/uv-k5-firmware-custom#100] [General]
When working a repeater and an external mike is used, it is impossible to PTT aslong as the repeater is still transmitting. The LED on top becomes blue when PTT then. This looks as it is a software bug. Has anyone noticed this? Is this fixable?

**Comment by OE1MWW** [comment]
Think this is not a 'bug', it's a 'feature' ;-)
Look at Menu 11, if it's ON
_BusyCL - busy channel lockout, blocks radio from transmitting because signal is being received (with BUSY on screen while PTT is pressed)__

**Reply by ldogge** [reply]
It is not related to that setting, because it only happens on repeaters, not in simplex. BCL is set to off. This must be a software bug. 
Because it also only happens when an external mike is in use.

**Reply by ldogge** [reply]
Also, why does the led turn blue and the USB charge icon appears on the LCD at that moment?

**Comment by ErikS-web** [comment]
Just read the WIKI under [`External Button PTT`](https://github.com/egzumer/uv-k5-firmware-custom/wiki/Button-functions#ptt-1)

It is NOT a software bug. It a hardware issue.
Workaround is to press the Internal PTT if you want to interrupt a received signal (like a repeater)

**Reply by ldogge** [reply]
This is not a hardware bug because in simplex it does work. Internal wiring for simplex or repeater is exactly the same. using the PTT on the radio is not a solution. I use an external one for reasons. It is deffenitly a bug in Quanshengs firmware.

**Comment by ldogge** [comment]
Maybe somebody can have a good look at the source code? I am pretty sure it is not an hardware bug.
It is a bit of a shame to label this as wont fix. It is a serious issue.  s a technical person I am only trying to help. 
Why close this topic so fast and already?

**Reply by ErikS-web** [reply]
This is a hobby project for us. You can also help investigate yourself.
Like...
Try other original or modified firmware to find out if that works.
Also check which hardware board is internal to your radio.
And try different external microphones

**Reply by ErikS-web** [reply]
In Simplex same result as explained on Wiki. With Latest Official release 0.17 on a UV-K5(8)

**Reply by ErikS-web** [reply]
Same result with official firmware k5_v2.01.31_publish.bin

**Reply by ErikS-web** [reply]
Has there been any progress with testing the hardware?

Please also take some photos.

**Comment by egzumer** [comment]
This is not a software bug. I don't know how you got different result in repeaters and simplex, from software side there is no difference. Repeater communication is simplex communication just with a shift. I can't transmit in "simplex" while receiving is active. This is with a radio with board v1.4. Just tested Radio with board v1.6 and there is no such problem. It seems to be fixed there. Also started measuring things. There are differences in the external PTT circuit in the radio.

**Comment by ErikS-web** [comment]
Opening the radio is simple ([demo](https://www.youtube.com/watch?v=gtwo1ak4nyE)) and look on PCB for revision level

This one is V1.4

![1 4](https://github.com/egzumer/uv-k5-firmware-custom/assets/148579604/5de6e0af-cbab-4292-a260-c501cda66962)

**Comment by ErikS-web** [comment]
Some adjustments have been made in the area at the top right. See the circuit board for revision level.

![Knipsel](https://github.com/egzumer/uv-k5-firmware-custom/assets/148579604/163cad9f-1808-470c-9292-ec41de0670a0)

With the new radio (with 1.6 board), the `external PTT` works the same as the `internal PTT`.
So if you want to solve this problem, the new scheme will have to be analyzed and what the adjustments are.
Fortunately, this is a hobby, so give it a try.

Or you can buy a new radio as a quick solution.

**Comment by perrymou** [comment]
I get this too. The external mic will not key the TX unless the squelch is closed. BusyCL setting makes no difference. It's not such a big deal to me, to use the PTT button on the radio instead (that works fine).
This firmware is awesome. I haven't been able to put the radio down since I flashed it. Thank you egzumer, fantastic job :)

**Comment by ErikS-web** [comment]
The pointed part is a new power IC. See Details: [New Power Chip "AWGAY" PCB V1.5](https://github.com/ludwich66/Quansheng_UV-K5_Wiki/wiki/Parts_Details#details-new-power-chip-awgay-pcb-v15)
